# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1433265884 -7200
#      Tue Jun 02 19:24:44 2015 +0200
# Branch stable
# Node ID 0bb456e9423bfbe68b16815a28bde74548e09cea
# Parent  ede7224d062fa7990afc56b0289124567bf0cc45
Allow interactive annotations in other figures than the currentfigure.

* Canvas.cc (Canvas::canvasMouseReleaseEvent): prepend the figure handle argument when calling annotation function.

diff --git a/libgui/graphics/Canvas.cc b/libgui/graphics/Canvas.cc
--- a/libgui/graphics/Canvas.cc
+++ b/libgui/graphics/Canvas.cc
@@ -789,23 +789,24 @@ Canvas::canvasMouseReleaseEvent (QMouseE
           if (w)
             {
               Matrix bb = figObj.get ("position").matrix_value ();
               bb(0) = m_mouseAnchor.x () / bb(2);
               bb(1) = 1.0 - (m_mouseAnchor.y () / bb(3));
               bb(2) = (event->x () - m_mouseAnchor.x ()) / bb(2);
               bb(3) = (m_mouseAnchor.y () - event->y ()) / bb(3);
 
-              octave_value_list props = ovl("textbox", bb);
+              octave_value_list props = ovl ("textbox", bb);
 
               annotation_dialog anno_dlg (w, props);
 
               if (anno_dlg.exec () == QDialog::Accepted)
                 {
                   props = anno_dlg.get_properties ();
+                  props.prepend (figObj.get_handle ().as_octave_value ());
 
                   octave_link::post_event (this, &Canvas::annotation_callback,
                                            props);
                 }
             }
         }
     }
   m_rectMode = false;
